edxval/api.py

```diff
@@ -733,6 +733,30 @@ def get_videos_for_course(course_id, sort_field=None, sort_dir=SortDirection.asc
     )
 
 
+@cachetools.func.ttl_cache(maxsize=None, ttl=settings.TRANSCRIPT_LANG_CACHE_TIMEOUT)
+def get_transcript_languages(course_id, provider_type):
+    """
+    Returns a list of languages for which transcripts are available for a course
+
+    Args:
+        course_id (str): course id
+        provider_type (str): transcript provider type
+
+    Returns:
+        (list): A list of language codes
+    """
+    transcript_languages = set()
+    video_transcripts = VideoTranscript.objects.filter(
+        video_id__in=CourseVideo.objects.filter(course_id=course_id).values_list(
+            "video_id"
+        )
+    )
+    for transcript in video_transcripts:
+        if transcript.provider == provider_type:
```

Comment:
Can't we do something like 
```
course_video_ids = CourseVideo.objects.filter(course_id=course_id, hidden=False).values_list('video__id')
transcripts = VideoTranscript.objects.filter(video__id_in=course_video_ids, provider=provider).values_list('language')
```

Reply:
Sure, good point. We should filter transcripts based on the provider type at db level rather than iterating through list. Let me add it